google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -417,7 +414,35 @@ public Page<Blob> list(String bucket, BlobListOption... options) {
 
   @Override
   public Bucket update(BucketInfo bucketInfo, BucketTargetOption... options) {
-    return todo();
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+    GrpcCallContext grpcCallContext = GrpcRequestMetadataSupport.create(optionsMap);
+    com.google.storage.v2.Bucket bucket = codecs.bucketInfo().encode(bucketInfo);
+    UpdateBucketRequest.Builder bucketRequestBuilder =
+        UpdateBucketRequest.newBuilder().setBucket(bucket);
+    bucketRequestBuilder.setUpdateMask(
+        FieldMask.newBuilder()
+            .addAllPaths(
+                bucket.getAllFields().entrySet().stream()
```

Comment:
Not sure if this is working as intended with fieldMask until integration tests are introduced: @BenWhitehead

Reply:
Addressed offline.